File: flutter_client/lib/shared/models/geo_fence.dart
/// 电子围栏模型
class GeoFence {
  final String id;
  final String elderId;
  final String? elderName;
  final double centerLatitude;
  final double centerLongitude;
  final int radius;
  final bool isEnabled;
  final String createdBy;
  final DateTime createdAt;
  final DateTime updatedAt;

  const GeoFence({
    required this.id,
    required this.elderId,
    this.elderName,
    required this.centerLatitude,
    required this.centerLongitude,
    required this.radius,
    required this.isEnabled,
    required this.createdBy,
    required this.createdAt,
    required this.updatedAt,
  });

  factory GeoFence.fromJson(Map<String, dynamic> json) {
    return GeoFence(
      id: json['id'] as String,
      elderId: json['elderId'] as String,
      elderName: json['elderName'] as String?,
      centerLatitude: (json['centerLatitude'] as num).toDouble(),
      centerLongitude: (json['centerLongitude'] as num).toDouble(),
      radius: json['radius'] as int,
      isEnabled: json['isEnabled'] as bool,
      createdBy: json['createdBy'] as String,
      createdAt: DateTime.parse(json['createdAt'] as String),
      updatedAt: DateTime.parse(json['updatedAt'] as String),
    );
  }

  /// 获取格式化的更新时间
  String get formattedUpdatedAt {
    final local = updatedAt.toLocal();
    return '${local.year}-${local.month.toString().padLeft(2, '0')}-${local.day.toString().padLeft(2, '0')} '
        '${local.hour.toString().padLeft(2, '0')}:${local.minute.toString().padLeft(2, '0')}';
  }

  /// 格式化半径显示
  String get radiusDisplay {
    if (radius >= 1000) {
      return '${(radius / 1000).toStringAsFixed(1)} 公里';
    }
    return '$radius 米';
  }
}

File: CareForTheOld/Models/Entities/GeoFence.cs
using CareForTheOld.Common.Constants;

namespace CareForTheOld.Models.Entities;

/// <summary>
/// 电子围栏（安全区域）实体
/// </summary>
public class GeoFence
{
    public Guid Id { get; set; }

    /// <summary>
    /// 监控的老人用户ID
    /// </summary>
    public Guid ElderId { get; set; }

    /// <summary>
    /// 围栏中心纬度
    /// </summary>
    public double CenterLatitude { get; set; }

    /// <summary>
    /// 围栏中心经度
    /// </summary>
    public double CenterLongitude { get; set; }

    /// <summary>
    /// 围栏半径（米），默认500米
    /// </summary>
    public int Radius { get; set; } = AppConstants.GeoFence.DefaultRadiusMeters;

    /// <summary>
    /// 是否启用
    /// </summary>
    public bool IsEnabled { get; set; } = true;

    /// <summary>
    /// 创建者用户ID（子女）
    /// </summary>
    public Guid CreatedBy { get; set; }

    /// <summary>
    /// 创建时间
    /// </summary>
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    /// <summary>
    /// 更新时间
    /// </summary>
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    // 导航属性
    public User Elder { get; set; } = null!;
    public User Creator { get; set; } = null!;
}